services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java

```diff
@@ -559,6 +560,7 @@ private <T> void addQueryFilterParameters(RealizationsFilter filter, TypedQuery<
     if (filter.getFromDate() != null && filter.getToDate() != null) {
       query.setParameter(FROM_DATE_PARAM_NAME, filter.getFromDate());
       query.setParameter(TO_DATE_PARAM_NAME, filter.getToDate());
+      query.setParameter("type", IdentityType.USER);
```

Comment:
IMO, you should add the type in the filter

Reply:
done; IdentityType.getType() return USER if the input is null .